src/pages/Team/Team.tsx

```diff
@@ -0,0 +1,192 @@
+
+import './style.css';
+
+const Team = (): JSX.Element => {
+  const codes = useSelector((state: RootState) => state.income.codes);
+  const dbIncomes = useSelector((state: RootState) => state.income.dbIncomes);
+  const registry = useSelector((state: RootState) => state.income.registry);
+  const importDates = useSelector((state: RootState) => state.income.importDates);
+
+  // const [editedData, setEditedData] = useState<string>('income');
+  const [displayedIncome, setDisplayedIncome] = useState<IncomesDb[]>([]);
+  
+  const [event, setEvent] = useState<string>('');
+  const [currentTeamRegistry, setCurrentTeamRegistry] = useState<Person[]>([]);
+  const [incomesByCode, setIncomeByCode] = useState<IncomesWithImportDate[] | null>([]); 
+  
+  const [incomesSC, setIncomesSC] = useState<number | null>(null);
+
+  const location = useLocation();
+  const currentTeam = location.pathname.split('/')[2];
+
+  const [rows, setRows] = useState<IncomesDb[]>([]);
+
+  const [useDate, setUseDate] = useState<boolean>(true);
+
+  const [selectedDate, setSelectedDate] = useState<Date>(new Date());
+
+  const handleDateChange = (date: Date | null) => {
+    date && setSelectedDate(date);
+  };
+
+  useEffect(() => {
+    
+    const teamtRegistry = registry && registry[currentTeam];
+    teamtRegistry && registry && setCurrentTeamRegistry(registry[currentTeam]);
+    const incomesToDisplay = dbIncomes && dbIncomes.filter(income => income.team === currentTeam);
+    setIncomeByCode(incomesToDisplay);
+  },[registry]);
+
+  useEffect(() => {
+    const helper = incomesByCode?.length ? (incomesByCode.map((el, index) => {
+      return ({
+        ...el,
+        lp: index + 1,
+        id: index,
+      });
+    })) : ([]);
+    setRows(helper);
+    const sum = incomesByCode && incomesByCode
+      .filter(income => income.event === 'SC')
+      .reduce((sum: number, income) => sum + income.cash ,0); 
+
+    setIncomesSC(sum);
+  },[incomesByCode]);
+
+
+  const columns = [
+    { field: 'lp', headerName: 'LP', width: 80, },
+    { field: 'name', headerName: 'IMIĘ', width: 150, },
+    { field: 'surname', headerName: 'NAZWISKO', width: 150, },
+    { field: 'cash', headerName: 'KWOTA', width: 150, },
+    { field: 'title', headerName: 'TYTUŁ', width: 800, },
+    { field: 'importDate', headerName: 'DATA', width: 150, },
+  ];
+
+  useEffect(() => {
+    const filteredIncomes = rows && rows.filter(i => {
+      if (useDate && selectedDate && i.importDate !== selectedDate.toLocaleString().split(',')[0]) return false;
+      if (event !== '' && i.event !== event && event !== 'unAssigned') return false;
+      if (event !== '' && event !== 'unAssigned' && !(i.name && i.surname && i.event && i.importDate && i.team && i.title && i.year && i.cash)) return false;
+      if (event !== '' && event === 'unAssigned' && i.name && i.surname && i.event && i.importDate && i.team && i.title && i.year && i.cash) return false;
```

Comment:
Very long code line we should split to clearly version  
  
DO:

if (a

&& b

&& c)

DON'T:  
if(a && b && c)

Eslint guards the line lenght - we should always resolve all eslint's problems.